clang/lib/Sema/SemaDecl.cpp

```diff
@@ -6970,6 +6970,18 @@ static void checkAttributesAfterMerging(Sema &S, NamedDecl &ND) {
         }
       }
     }
+
+    for (unsigned int I = 0; I < FD->getNumParams(); ++I) {
```

Comment:
We would also want to diagnose this for member functions where lifetimebound is not attached to a param but to the function type.
```cpp
struct S {
  void member() [[clang::lifetimebound]];
}
```

Reply:
Makes sense!
Added a test for that with FIXME to fix in another pull request.